nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py

```diff
@@ -85,24 +139,65 @@ def load_sharded_metadata_torch_dist(checkpoint_dir: Union[Path, TarPath], torch
     return state_dict
 
 
+def load_sharded_pickle_extra_state_scale(dir: Union[Path, TarPath]):
+    scales = []
+    layer_number = 0
+
+    while pt_file_list := list(dir.glob(f'shard_{layer_number}_*.pt')):
+        pt_file = pt_file_list[0]
+        with pt_file.open('rb') as checkpoint_file:
```

Comment:
Hey @mikolajblaz. Could you please help us with the recommended way of reading extra states? For example, can we somehow avoid double `torch.load` and that `dictionary.seek(0)` call?

(Background is that we need to process extra states to get FP8 scaling factors for direct FP8 TRT-LLM deployment.)

Reply:
We discussed this and it is fine to import from `megatron.core` here, especially in the light of https://github.com/NVIDIA/NeMo/pull/10238, that's a minor detail anyway.